eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -453,13 +501,38 @@ private void scheduleServerEndpointTask(String zone) {
                 clientConfig.getEurekaServiceUrlPollIntervalSeconds(),
                 clientConfig.getEurekaServiceUrlPollIntervalSeconds(), TimeUnit.SECONDS);
 
+        eurekaTransport.bootstrapResolver = EurekaHttpClients
+                .newBootstrapResolver(clientConfig, applicationInfoManager.getInfo());
+
+        eurekaTransport.transportClientFactory = EurekaHttpClients
+                .newTransportClientFactory(clientConfig, applicationInfoManager.getInfo());
+
+        if (clientConfig.shouldRegisterWithEureka()) {
```

Comment:
Correct. There should be no external dependencies (e.g. network calls) needed for the new transport init, they should all be lazy.